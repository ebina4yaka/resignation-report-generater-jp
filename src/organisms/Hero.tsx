import { ReactElement } from 'react'
import Image from 'next/image'
import Container from '@material-ui/core/Container'
import Typography from '@material-ui/core/Typography'
import { createStyles, makeStyles } from '@material-ui/core/styles'
import HeroButtons from '../molecules/HeroButtons'

const useStyles = makeStyles((theme) =>
  createStyles({
    root: {
      padding: theme.spacing(8, 0, 4),
    },
    buttons: {
      marginTop: theme.spacing(4),
      marginBottom: theme.spacing(2),
    },
    contents: {
      display: 'flex',
      flexDirection: 'column',
      alignItems: 'center',
      justifyContent: 'center',
      textAlign: 'center',
    },
    image: {
      margin: theme.spacing(2),
    },
  })
)

export default function Hero(): ReactElement {
  const classes = useStyles()
  return (
    <div className={classes.root}>
      <Container maxWidth="md">
        <div className={classes.contents}>
          <Typography
            component="h1"
            variant="h4"
            align="center"
            color="textPrimary"
            gutterBottom
          >
            退職届ジェネレーター
          </Typography>
          <Typography
            component="h2"
            variant="h5"
            align="center"
            color="textSecondary"
            paragraph
          >
            内容証明郵便で叩きつけよう！
          </Typography>
          <div className={classes.image}>
            <Image src="/top.png" width={200} height={200} />
          </div>
          <div className={classes.buttons}>
            <HeroButtons />
          </div>
          <Typography
            component="h3"
            align="center"
            color="textSecondary"
            paragraph
          >
            ※Word形式(.docx)での出力に対応しています。
          </Typography>
        </div>
      </Container>
    </div>
  )
}
