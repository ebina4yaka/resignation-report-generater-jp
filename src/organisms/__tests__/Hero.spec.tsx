import React from 'react'
import renderer from 'react-test-renderer'
import Hero from '../Hero'

jest.mock('../../molecules/HeroButtons', () => 'MockedHeroButtons')
jest.mock('next/image', () => 'MockedImage')

test('Snapshot test for Hero', () => {
  const tree = renderer.create(<Hero />).toJSON()
  expect(tree).toMatchSnapshot()
})
