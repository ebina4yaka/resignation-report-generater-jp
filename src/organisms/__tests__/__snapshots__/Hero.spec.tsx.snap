// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test for Hero 1`] = `
<div
  className="makeStyles-root-1"
>
  <div
    className="MuiContainer-root MuiContainer-maxWidthMd"
  >
    <div
      className="makeStyles-contents-2"
    >
      <h1
        className="MuiTypography-root MuiTypography-h4 MuiTypography-colorTextPrimary MuiTypography-gutterBottom MuiTypography-alignCenter"
      >
        退職届ジェネレーター
      </h1>
      <h2
        className="MuiTypography-root MuiTypography-h5 MuiTypography-colorTextSecondary MuiTypography-paragraph MuiTypography-alignCenter"
      >
        内容証明郵便で叩きつけよう！
      </h2>
      <div
        className="makeStyles-image-3"
      >
        <MockedImage
          height={200}
          src="/top.png"
          width={200}
        />
      </div>
      <MockedHeroButtons />
      <h3
        className="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary MuiTypography-paragraph MuiTypography-alignCenter"
      >
        ※Word形式(.docx)での出力に対応しています。
      </h3>
    </div>
  </div>
</div>
`;
